### PR TITLE
Hometask4

### PR DESCRIPTION
**Pull request by bogdanbasii**
Phonebook. Made it a bit strange but it works. I hope you will evaluate it as acceptable

### CODE REVIEW

File: hometask 4.py [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+    elif command == 'del':
+        print(phonebook)
+        del phonebook[input('Enter a name to delete: ')]
+        print('Contact was deleted')
+    elif command == 'list':
+        list = (phonebook.items())
+        print(list)
+    elif command == 'show':
+        name=input('Enter a name:')
+        print(phonebook[name])
```
Comment (klymenok) [inline hometask 4.py]: It is better to check if key exists before accessing the value

File: hometask 4.py [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+        print(len(phonebook))
+    elif command == 'add':
+         for i in range(1):
+             name=input('Enter a name:')
+             number=input('Enter a number:')
+             phonebook[name]=number
+             print('New contact was added')
+    elif command == 'del':
+        print(phonebook)
+        del phonebook[input('Enter a name to delete: ')]
```
Comment (klymenok) [inline hometask 4.py]: It is better to check if key exists before deleting it

File: hometask 4.py [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+while True:
+    user_input = input('Enter a command:')
+    split_input = user_input.split()
+
+    command = split_input[0]
+
+    if command == 'stats':
+        print(len(phonebook))
+    elif command == 'add':
+         for i in range(1):
```
Comment (klymenok) [inline hometask 4.py]: What this `for-in` for? Why do you need this iteration?

File: hometask 4.py [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+             name=input('Enter a name:')
+             number=input('Enter a number:')
+             phonebook[name]=number
```
Comment (klymenok) [inline hometask 4.py]: There shouldn't exist an ability to override records. Please, make additional check here

**Review by klymenok — CHANGES_REQUESTED** [review]

**Review by klymenok — CHANGES_REQUESTED** [review]
Please, check other comments above

File: hometask 4.py (hunk 4 above)
Comment (bogdanbasii) [reply hometask 4.py]: There is no such lines in code anymore. Please revise.

File: hometask 4.py (hunk 1 above)
Comment (bogdanbasii) [reply hometask 4.py]: Check is integrated in code. Please revise.

File: hometask 4.py [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+    elif command == 'add':
+        name=input('Enter a name:')
+        if name in phonebook.keys():
+            print("Fault. There is such name in a phonebook.")
+
+        else:
+            print("There is no such name in a phonebook. Continue.")
+
+        number=input('Enter a number:')
+        phonebook[name]=number
```
Comment (klymenok) [inline hometask 4.py]: There shouldn't exist an ability to override records. Your check above just informs that the name is already in phonebook, but then you anyway override it. Move it inside of `else` condition

**Review by klymenok — CHANGES_REQUESTED** [review]

File: hometask 4.py [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+    if command == 'stats':
+        print(len(phonebook))
+    elif command == 'add':
+        name=input('Enter a name:')
+        if not name in phonebook.keys():
+           number = input('Enter a number:')
+           phonebook[name]=number
+           print('New contact was added')
+        else:
+            name in phonebook.keys()
```
Comment (klymenok) [inline hometask 4.py:30]: This line can be removed since it has no effect

**Review by klymenok — APPROVED** [review]
Looks good, one line can be removed

**Pull request merged**